components/style/properties/declaration_block.rs

```diff
@@ -89,13 +89,21 @@ impl PropertyDeclarationBlock {
 
                 // Step 2.2.2 & 2.2.3
                 match declaration {
-                    Some(&(ref declaration, _importance)) => list.push(declaration),
+                    Some(&(ref declaration, importance)) => {
+                        // If there is an longhand with important, we don't
+                        // serialize it as a shorthand.
+                        if importance.important() {
```

Comment:
We should still serialize it with `!important` if all of them have `!important`, so this is not enough.

Please add a test for it.